toolchain/check/import_cpp.cpp

```diff
@@ -220,6 +223,25 @@ static auto ImportFunctionDecl(Context& context, SemIR::LocId loc_id,
   auto decl_id = AddPlaceholderInst(
       context, SemIR::LocIdAndInst(Parse::NodeId::None, function_decl));
```

Comment:
Can we move this to after we create the return type?
If so, we could merge the condition that check the return type is supported and the condition switches over the return type.